extension/commands/associateIdentityWithNode.ts

```diff
@@ -168,28 +168,31 @@ export async function associateIdentityWithNode(replace: boolean = false, enviro
                 break;
             }
 
-            const items: string[] = otherNodes.map((_node: FabricNode) => {
-                return `Yes, ${_node.name}`;
-            });
+            const yesnoPick: string = await UserInputUtil.showQuickPickYesNo('Do you want to associate the same identity with another node?');
 
-            items.push('No');
-            const nodeName: string = await UserInputUtil.showQuickPick('Do you want to associate the same identity with another node?', items) as string;
-
-            if (!nodeName || nodeName === 'No') {
+            if (!yesnoPick || yesnoPick === UserInputUtil.NO) {
                 askAgain = false;
             } else {
-                const foundNode: FabricNode = otherNodes.find((_node: FabricNode) => nodeName.endsWith(_node.name));
+                    const nodes: IBlockchainQuickPickItem<FabricNode>[] = await UserInputUtil.showFabricNodeQuickPick('Choose the nodes you wish to associate with this identity', environmentRegistryEntry.name, [], false, true) as IBlockchainQuickPickItem<FabricNode>[];
```

Comment:
this doesn't work as it shows you all the nodes so you are shown all the ones that already have identities and the one that you just assigned an identity too. You need to use the nodes in `otherNodes`.

Reply:
Fixed